From 806fc43f29a2351c921f82ffb82f1245b3b30ccc Mon Sep 17 00:00:00 2001
From: Zorro Liu <lyx@rock-chips.com>
Date: Mon, 13 Nov 2017 10:29:47 +0800
Subject: [PATCH] driver: input: sensor: client is always null, we should not
 use it

[    3.427327] [<ffffff8008520c18>] dev_driver_string+0x0/0x34
[    3.432895] [<ffffff800852157c>] _dev_info+0x74/0x94
[    3.437860] [<ffffff8008690988>] sensor_register_slave+0x80/0x8c
[    3.443858] [<ffffff8009052874>] gsensor_mma8452_init+0x24/0x2c
[    3.449776] [<ffffff80080831f0>] do_one_initcall+0x178/0x194
[    3.455437] [<ffffff8009020ddc>] kernel_init_freeable+0x1a4/0x25c
[    3.461531] [<ffffff8008ae9940>] kernel_init+0x10/0xf8
[    3.466666] [<ffffff80080828d0>] ret_from_fork+0x10/0x40

Change-Id: I16e608e82c4da098b6d7b42fa7c3a2dc864724f6
Signed-off-by: Zorro Liu <lyx@rock-chips.com>
---
 drivers/input/sensors/sensor-dev.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/input/sensors/sensor-dev.c b/drivers/input/sensors/sensor-dev.c
index 72a7800cb80d..77ea439de115 100644
--- a/drivers/input/sensors/sensor-dev.c
+++ b/drivers/input/sensors/sensor-dev.c
@@ -1302,13 +1302,13 @@ int sensor_register_slave(int type, struct i2c_client *client,
 	struct sensor_operate *ops = get_sensor_ops();
 
 	if ((ops->id_i2c >= SENSOR_NUM_ID) || (ops->id_i2c <= ID_INVALID)) {
-		dev_err(&client->dev, "%s:%s id is error %d\n", __func__, ops->name, ops->id_i2c);
+		printk(KERN_ERR "%s:%s id is error %d\n", __func__, ops->name, ops->id_i2c);
 		return -1;
 	}
 	sensor_ops[ops->id_i2c] = ops;
 	sensor_probe_times[ops->id_i2c] = 0;
 
-	dev_info(&client->dev, "%s:%s,id=%d\n", __func__, sensor_ops[ops->id_i2c]->name, ops->id_i2c);
+	printk(KERN_INFO "%s:%s,id=%d\n", __func__, sensor_ops[ops->id_i2c]->name, ops->id_i2c);
 
 	return result;
 }
@@ -1321,10 +1321,10 @@ int sensor_unregister_slave(int type, struct i2c_client *client,
 	struct sensor_operate *ops = get_sensor_ops();
 
 	if ((ops->id_i2c >= SENSOR_NUM_ID) || (ops->id_i2c <= ID_INVALID)) {
-		dev_err(&client->dev, "%s:%s id is error %d\n", __func__, ops->name, ops->id_i2c);
+		printk(KERN_ERR "%s:%s id is error %d\n", __func__, ops->name, ops->id_i2c);
 		return -1;
 	}
-	dev_info(&client->dev, "%s:%s,id=%d\n", __func__, sensor_ops[ops->id_i2c]->name, ops->id_i2c);
+	printk(KERN_INFO "%s:%s,id=%d\n", __func__, sensor_ops[ops->id_i2c]->name, ops->id_i2c);
 	sensor_ops[ops->id_i2c] = NULL;
 
 	return result;
-- 
2.35.3

